src/vs/editor/common/model/textModel.ts

```diff
@@ -1508,7 +1502,15 @@ export class TextModel extends Disposable implements model.ITextModel {
 						let lineNumber = fromLineNumber + i;
 						newLines[lineNumber - fromLineNumber] = this.getLineContent(lineNumber);
 					}
-					rawContentChanges.push(new ModelRawLinesInserted(spliceLineNumber + 1, startLineNumber + insertingLinesCnt, newLines));
+
+					rawContentChanges.push(
```

Comment:
I think there is also something wrong here. `injectedText` contains injected text in the `decorationsWithInjectedTextInEditedRange` which covers lines from `firstEditLineNumber` up to `lastEditLineNumber`. It does not cover new lines. You can think the case is that I paste `\n\n\n\n\n\n\n` in the middle of a line that contains injected text, but before the injected text.

Reply:
Right. In my manual test, it worked since the range of the injected text spanned all the lines before. So when inserting `\n\n\n\n\n\n\n\n\n\n`, the range got extended automatically and the node was still in the editedRange, even though the injected text moved by 10 lines.

I'm just noticing that injected texts might be reported that are way before (when using `before: { ... }`) or after the edited range. Thus, the `takeWhile` approach would have failed - a `skipUntil` would also be needed.

Reply:
That could be worked around using e.g. `takeWhile((item) => item.lineNumber < firstEditLineNumber)`.

Reply:
Fixed. Though I got a little bit lost in all the constants that are defined here.